app/main.py

```diff
@@ -1,3 +1,23 @@
+import functools
+
+
 def cache(func):
-    # Write your code here
-    pass
+    saved_results = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if func.__name__ in saved_results:
```

Comment:
you can write this without func name, store args as key, and a result of the func as value

Reply:
hello, sorry for delay, i have proceed my learning, started new topic and accidentally forgot about this task. Could you please explain specifically how to avoid using func.__name__. The cache decorator won't work with different functions if i not store the func_name as a key.
ps: The issue about code duplication is completely clear. It's fixed

Reply:
better write in slack)